logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -173,11 +173,16 @@ class ConversationEventReceiver(
                 if (isSenderVerified(message.content.messageId, message.conversationId, message.senderUserId))
                     messageRepository.markMessageAsDeleted(messageUuid = message.content.messageId, conversationId = message.conversationId)
                 else kaliumLogger.i(message = "Delete message sender is not verified: $message")
-            is MessageContent.DeleteForMe ->
+            is MessageContent.DeleteForMe -> {
                 if (message.conversationId == conversationRepository.getSelfConversationId())
```

Comment:
> The fact that this was working on AR but not other platforms, means when the AR client receives `DeleteForMe` request from any conversation (not only self conversation), it's processed normally. This can be a vulnerability. If we have 3 users A, B, C in a group chat and A sends a `DeleteForMe` request with the userID of B the deleted message will be deleted from all of B clients but not A, C clients. I suggest we verify the conversation which is `DeleteFormMe` requests are received and do to delete if and only if the conversation is self; otherwise it should be ignored.

Ohhh my bad, i see you already did take care of it